topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java

```diff
@@ -65,6 +65,31 @@ public ActorFuture<TopologyChangeResult> applyOperations(final TopologyChangeReq
     return future;
   }
 
+  @Override
+  public ActorFuture<TopologyChangeResult> simulateOperations(final TopologyChangeRequest request) {
+    final ActorFuture<TopologyChangeResult> future = executor.createFuture();
+    executor.run(
```

Comment:
:question: Can we reuse most of the code in `applyOperations`? Here there is a risk of different behavior in simulation vs actual apply, especially when making changes in future.

Reply:
Yeah good point, I didn't go far enough the first time. See https://github.com/camunda/zeebe/pull/15170/commits/33bbd719278f93f0488de9becb09692ea2bc12e5, this should be better now,